### ISSUE TITLE
Add ability to filter by namespace

### ISSUE THREAD

**Issue by olevitt** [opencost/opencost#1772]
**Describe the solution you'd like**
At the moment the `allocation/compute` endpoint only allows full cluster-wide data retrieval which may be slow and resource-intensive when dealing with big clusters.  
I think it would be interesting to add a filter for that query to only retrieve data from a single namespace which is a common use-case in shared clusters.

**Comment by mattray** [comment]
To clarify, not aggregating by namespaces, but actually starting the query pinned to a particular namespace?

**Comment by olevitt** [comment]
Exactly. I saw that aggregating is already implemented but this is about getting only the data (non aggregated) relevant to a specific namespace

**Comment by mattray** [comment]
I think this would make a great PR if you're interested in taking it on.

**Comment by olevitt** [comment]
I'm not sure I will be able to do a PR as I'm quite new to golang and may not have enough time to learn enough of it. I may give it a try anyway but if anyone want to handle it, feel free to do so without waiting for me :smiley:

**Comment by mattray** [comment]
@AjayTripathy is this an enhancement that would be useful in the context of the UI in Kubecost? The API enhancement would be good, but if it has a practical value to Kubecost it might be worth getting on their roadmap. I don't believe OpenCost's UI would benefit from this currently.

**Comment by AjayTripathy** [comment]
Hmm... @mattray filtering by ns works in kubecost. https://docs.kubecost.com/apis/apis-overview/filters-api#using-filters and I think it should work in opencost and just isn't documented. Can you try the same syntax on opencost and let me know? If it doesn't work I think it should be quick to support.

**Comment by olevitt** [comment]
`http://localhost:9003/allocation/compute?window=today&filter=namespace:%22opencost%22` does not appear to filter (version `kubecost-cost-model:prod-1.101.2`)

**Comment by mattray** [comment]
@AjayTripathy confirming @olevitt's observation I can't get anything to filter either. Do we consider this a bug or a feature request?

**Comment by AjayTripathy** [comment]
Hmm. @mattray can you try the v1 filtering parameters? ie 
`&filterNamespaces="opencost"`

Either way I think this is important and we'll want to fix/implement the new API... @Adam-Stack-PM let's get this tracked.

**Comment by olevitt** [comment]
From my test (`kubecost-cost-model:prod-1.101.2`) 
```
http://localhost:9003/allocation/compute?window=today&filterNamespaces="opencost"
```
(or `http://localhost:9003/allocation/compute?window=today&filterNamespaces=opencost`)
still retrieves info from all namespaces

**Comment by mattray** [comment]
Yeah, nothing filters for me with v1 or v2

**Comment by Adam-Stack-PM** [comment]
@AjayTripathy, Can do. CC:@allyleonce

**Comment by Adam-Stack-PM** [comment]
@kwombach12, Getting this item on your radar for prioritization for v1.104.

**Comment by olevitt** [comment]
Hi ! Any news on this ?

**Comment by olevitt** [comment]
> @kwombach12, Getting this item on your radar for prioritization for v1.104.

any news on this feature ?

**Comment by kwombach12** [comment]
I'm still seeing this issue as well. I've filed a ticket with Kubecost ENG: https://kubecost.atlassian.net/browse/GTM-181. @olevitt @mattray Is there community interest in authoring a PR to resolve this? I'd also strongly suggest we update our docs to reflect the filtering capability once added.

**Comment by olevitt** [comment]
I tried to look at the code and submit a PR but I didn't get even close to it due to my lack of go knowledge. Sorry. 
I would happily test this feature if someone manage to write it tho

**Comment by zhaoyue722** [comment]
Hi! Any updates?

**Comment by AjayTripathy** [comment]
This isn't a priority for us at the moment but would still be interested in a PR here!
